.github/workflows/e2eCI.yaml

```diff
@@ -0,0 +1,56 @@
+---
+name: kotary e2e testing
+on: push
+jobs:
+  kind:
+    runs-on: ubuntu-latest
+    name: Test Kotary Operator on Kind cluster. Kube version
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes:
+          - 1.21
+          - 1.23
+          - 1.24
+          - 1.26
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+      
+      - name: Docker Build image
+        run: | 
+          docker build -t "ci/kotary:${{ matrix.kubernetes }}" .
```

Comment:
this is confusing. When multiple jobs will run together, they will override each other. Let's include a sha.